### ISSUE TITLE
Problem using more nodes/expanding nodes (currently 47 nodes)

### ISSUE THREAD

**Issue by hspjanssen** [node-red/node-red#3050]
### What are the steps to reproduce?
I am using Node Red for a year now, without any problems. I have about 12 flows, since I am working on my latest flow I receive the message "Deploy failed: no response from server", when I deploy the configuration. I did some further testing and I now figured out that when I remove one node from any other flows, I can add one node to any flow, when I then hit deploy, the configuration is "Successfully deployed". If I want to add two nodes, then I have to remove two nodes somewhere else. On advise of someone on noderedforum, I already increased the "apiMaxLength" to 10mb and 100mb, without results.

### What happens?
When I add one node to a flow and I hit the deploy button, I receive the mssage "Deploy failed: no response from server". When I remove one node (does not matter which one), I receive "Successfully deployed" when I hit the deployment button.

### What do you expect to happen?
I expect that I can extend with more nodes. I am using only default nodes, with one custom node (used once). I am now using 47 nodes in total. I also removed the custom node, but same problem.

### Please tell us about your environment:

- [ ] Node-RED version:
- [ ]1.3.5
- [ ] Node.js version:
- [ ] npm version:
- [ ] Platform/OS:
- [ ] Debian 10.9
- [ ] Browser:
- [ ] Chrome
- [ ] running in Docker:
- [ ] Yes

**Comment by knolleary** [comment]
What do the runtime logs show when you get the failed deploy?

**Comment by knolleary** [comment]
Also being actively discussed on the forum here https://discourse.nodered.org/t/problem-using-more-nodes-expanding-nodes-currently-47-nodes/47961

Please pick one place to discuss the issue rather than split it between the forum and here. You're talking to the same people in both places.

**Comment by hspjanssen** [comment]
> Also being actively discussed on the forum here https://discourse.nodered.org/t/problem-using-more-nodes-expanding-nodes-currently-47-nodes/47961
> 
> Please pick one place to discuss the issue rather than split it between the forum and here. You're talking to the same people in both places.

Ok, sorry, did not know that. See attached log, is that the one you are looking for?
[logging.txt](https://github.com/node-red/node-red/files/6758193/logging.txt)

**Comment by knolleary** [comment]
That is the right log. That shows the runtime is stopping all the nodes, but there's no indication that it is starting the new flow configuration. Is that all the log you have? Are there no error messages etc?

**Comment by hspjanssen** [comment]
Yes that is all there is in the log. As I am using Docker, I did a Docker attach to the container, write the log in the file and waited to abort the attach until the message "Deploy failed" or "Successfully deployed" popped up in the Webbrowser.

Hereby a log with a succesfully deployment and one with deploy failed.
[Deploy_failed.txt](https://github.com/node-red/node-red/files/6758383/Deploy_failed.txt)
[Successfully deployed.txt](https://github.com/node-red/node-red/files/6758384/Successfully.deployed.txt)

**Comment by hspjanssen** [comment]
I made a video of an example, see attachment.

https://user-images.githubusercontent.com/32012352/124358113-1a673380-dc1f-11eb-80cf-71cbb5093442.mp4

**Comment by hspjanssen** [comment]
Here another example including the associated logging.

https://user-images.githubusercontent.com/32012352/124359896-2f47c500-dc27-11eb-8057-07d83e935670.mp4

[Example2.txt.txt](https://github.com/node-red/node-red/files/6758659/Example2.txt.txt)

**Comment by hspjanssen** [comment]
Installed Node-Red on a LXC Debian container, export flows from old installation, import on new installation, works ok now. Has something to do with Docker, remove imaged and downloaded it again, without any difference.

Any idea why this is happening on Docker?

**Comment by normanr** [comment]
How large is the POST to /flows? When you changed apiMaxLength where did you do that? Did you configured the docker image to COPY settings.js?

**Comment by VincentSC** [comment]
Opened an issue in the node-red-docker projects, as the problem seems to be linked to Docker

**Comment by VincentSC** [comment]
Reacting to https://github.com/node-red/node-red-docker/issues/305#issuecomment-1046579006

Where in Node Red is the saving of an updated flow handled? Is there a list of files+functions? A post to `/flow` is all I know, and I want to avoid a situation where I miss a certain file, as I am more into C/C++ than JS.

**Comment by knolleary** [comment]
Closing as part of a tidy-up of the issue tracker — this has been inactive for a long time and much has changed in Node-RED since. If it's still relevant on a current version, please open a fresh issue with up-to-date details and we'll take a look. Thanks for raising it.
